[bug] static runtime is ignored for msvc compiler

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version: MSVC 19.31.30919.0
  * Conan version: 1.44.0
  * Python version: 3.9.7

### Steps to reproduce (Include if Applicable)
Attempt to build a package (those I tested used cmake under the hood) with `settings.compiler.runtime=static` using the latest msvc compiler. When using `settings.compiler.runtime_type=Debug` it should build with `/MTd`, however it actually uses `/MDd`. When using `settings.compiler.runtime_type=Release` it should build with `/MT`, however it actually uses `/MD`.

The `/MDd` or `/MD` flavor should only be used when the `settings.compiler.runtime` configuration is set to dynamic, otherwise it should use `/MT` or `/MTd`

It appears to be impossible (or I have not yet found a way) to convince conan to use the desired runtime.

When setting the compiler in the profile as "Visual Studio" I do however get an error (without any error message) that building the recipe failed.
Manually executing the command it attempts to execute reveals an error between the compiler version as detected and as specified.
```
CMake Error at conanbuildinfo.cmake:511 (message):
  Detected a mismatch for the compiler version between your conan profile
  settings and CMake:

  Compiler version specified in your conan profile: 17

  Compiler version detected in CMake: 19.31
```
However 17 is the highest compiler version conan knows of.

This can be worked around by adding `cmake.definitions['CONAN_DISABLE_CHECK_COMPILER'] = True` in the recipe, this however is not present in official recipes I use, which will consequently fail to build.

Is there any way to have this specified in the profile / globally?

PS: This is turning out to be a major headache :(

I am trying to run conan install --build=missing, so yes it is attempting to build one dependency as that does not seem to have a prebuilt version for Visual Studio version 17.

```
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=Visual Studio
compiler.cppstd=17
compiler.runtime=MTd
compiler.version=17
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

zlib/1.2.11: Configuring sources in <redacted>\.conan\data\zlib\1.2.11\_\_\source
Downloading zlib-1.2.11.tar.gz completed [593.46k]

zlib/1.2.11: Copying sources to build folder
zlib/1.2.11: Building your package in <redacted>\.conan\data\zlib\1.2.11\_\_\build\3682397a88110bb423c0b4c7c75a818b7fdf9bd9
zlib/1.2.11: Generator cmake created conanbuildinfo.cmake
zlib/1.2.11: Aggregating env generators
zlib/1.2.11: Calling build()
-- Selecting Windows SDK version 10.0.20348.0 to target Windows 10.0.19042.
-- The C compiler identification is MSVC 19.31.30919.0
-- The CXX compiler identification is MSVC 19.31.30919.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Preview/VC/Tools/MSVC/14.31.30919/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Preview/VC/Tools/MSVC/14.31.30919/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
CMake Error at conanbuildinfo.cmake:402 (message):
  Detected a mismatch for the compiler version between your conan profile
  settings and CMake:

  Compiler version specified in your conan profile: 17

  Compiler version detected in CMake: 19.31

  Please check your conan profile settings (conan profile show
  [default|your_profile_name])

  P.S.  You may set CONAN_DISABLE_CHECK_COMPILER CMake variable in order to
  disable this check.
Call Stack (most recent call first):
  conanbuildinfo.cmake:470 (conan_error_compiler_version)
  conanbuildinfo.cmake:591 (check_compiler_version)
  conanbuildinfo.cmake:141 (conan_check_compiler)
  CMakeLists.txt:5 (conan_basic_setup)


-- Configuring incomplete, errors occurred!
See also "<redacted>/.conan/data/zlib/1.2.11/_/_/build/3682397a88110bb423c0b4c7c75a818b7fdf9bd9/build_subfolder/CMakeFiles/CMakeOutput.log".
zlib/1.2.11:
zlib/1.2.11: ERROR: Package '3682397a88110bb423c0b4c7c75a818b7fdf9bd9' build failed
```

Even after changing those lines to make it so that it won't fail the compiler detection, it will not detect a C compiler either.

```
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Conan setting CPP STANDARD: 17 WITH EXTENSIONS OFF
CMake Error at conanbuildinfo.cmake:648 (message):
  This project seems to be plain C, but no compiler defined
Call Stack (most recent call first):
  conanbuildinfo.cmake:250 (conan_check_compiler)
  CMakeLists.txt:3 (conan_basic_setup)


-- Configuring incomplete, errors occurred!
```

--- a/conans/client/generators/cmake_common.py
+++ b/conans/client/generators/cmake_common.py
@@ -472,7 +472,7 @@ class CMakeCommonMacros:
                 # https://cmake.org/cmake/help/v3.14/variable/MSVC_VERSION.html
                 if(
                     # 1930 = VS 17.0 (v143 toolset)
-                    (CONAN_COMPILER_VERSION STREQUAL "17" AND NOT MSVC_VERSION EQUAL 1930) OR
+                    (CONAN_COMPILER_VERSION STREQUAL "17" AND NOT((MSVC_VERSION EQUAL 1930) OR (MSVC_VERSION GREATER 1930))) OR
                     # 1920-1929 = VS 16.0 (v142 toolset)
                     (CONAN_COMPILER_VERSION STREQUAL "16" AND NOT((MSVC_VERSION GREATER 1919) AND (MSVC_VERSION LESS 1930))) OR
                     # 1910-1919 = VS 15.0 (v141 toolset)

